src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,6 +51,26 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  // If the object.workspaces is null/[] (as global object), return it directly.
+  // If the workspace is specified, validate whether the object exists in the workspace.
+  private validateObjectExistInWorkspaces<T>(
```

Comment:
nit: the name of this function is a little bit confusing as it will not really just validate which implies returning true or false but it would modify the object if it exists in the workspace. perhaps this could be a return of true return if the object workspaces includes the workspace.

if this is the current standard of saved objects then please ignore.

Reply:
Thanks for your suggestion, I have updated the function name to `formatObjectForWorkspaceValidation`